src/util/file.cc

```diff
@@ -132,6 +132,14 @@ future<stat_data> file_stat(std::string_view name, follow_symlink follow) noexce
     return engine().file_stat(name, follow);
 }
 
+future<std::optional<struct group>> getgrnam(std::string_view name, char *buf, size_t buflen) {
```

Comment:
please `#include <grp.h>` in this source file so that it has access the complete definition of `group`. you could add `#include <grp.h>` right before `#include <sys/statvfs.h>`.